### ISSUE TITLE
minify of destructured props on es5 is better than es2015+

### ISSUE THREAD

**Issue by nkitku** [swc-project/swc#7694]
### Describe the bug

its not removing unused variable which are assigned using destructuring objects in es2015+ but in es5 do

### Input code

```typescript
{
  const { b } = { b: 10 }
}

{
  const a = { a: 10 }.a
}

{
  const { b } = { b: 10 };
  console.log(b)
}

{
  const a = { a: 10 }.a
  console.log(a)
}
```

### Config

```json
{
  "minify": true,
  "jsc": {
    "minify": {
      "compress": {
        "dead_code": true
      },
      "mangle": {
        "toplevel": true,
        "keep_classnames": false,
        "keep_fnames": false,
        "keep_private_props": false,
        "ie8": false,
        "safari10": false
      }
    },
    "parser": {
      "syntax": "typescript"
    },
    "target": "es2022"
  }
}
```

### Playground link

https://play.swc.rs/?version=1.3.70&code=H4sIAAAAAAAAA6vmUlBIzs8rLlGoVkhSqFWwBdFWCoYGCrVctVxc1XDpRLBUIkRKLxFVEkOvNVQqPydVLyc%2FXSNJE79pqKoTgaoB6dRd1ZkAAAA%3D&config=H4sIAAAAAAAAA32QzwqDMAyH7z6F9OzBeRp7GQk1Srf%2BI8nGRHz3tUyhMtmpzff90pAuVV0rZ7wZZ3WrhZ7YZHJnncolXUv9rRPRwUVC5oIlOiAMvQ4Dbi9tZm32Ngd%2BsnhskhAtvtAW0zfzQIy9tsDswWGeNYLln8T410YyLxBMZ4hnIYPXE8owAplLu6t9k6rYR0UgRiq%2FhWcv8E5EyRyRNZko6tAjQBNKTiB3bddlu1brB422nWuGAQAA

### Expected behavior

```js
console.log(10),console.log(10);
```

### Actual behavior

```js
{let{b:l}={b:10}}{let{b:l}={b:10};console.log(l)}console.log(10);
```

### Version

1.3.70

**Comment by kdy1** [comment]
This is expected. Not intentional, but it's quite hard to analyze.
Try it on terser playground

**Comment by nkitku** [comment]
@kdy1 yes you are right, but can't we treat it similarly as in es5, I think it will give better result for almost all cases

**Comment by kdy1** [comment]
We can, but will you work on it?
We don't have enough resources

**Comment by nkitku** [comment]
@kdy1 ah, ok, i can understand, yes, to be honest it's too difficult for me, but maybe i'll try in my spare time, but if you don't mind, i think this issue should be kept open may be as future plans as it may help to reduce the modern codebase even more
